uat/custom-components/client-mosquitto-c/src/GRPCControlServer.cpp

```diff
@@ -213,29 +217,75 @@ Status GRPCControlServer::CloseMqttConnection(ServerContext * context, const Mqt
         delete connection;
         return Status::OK;
     } catch (MqttException & ex) {
+        loge("CloseMqttConnection: exception during disconnecting\n");
         return Status(StatusCode::INTERNAL, ex.getMessage());
     }
 }
 
 Status GRPCControlServer::PublishMqtt(ServerContext * context, const MqttPublishRequest * request, MqttPublishReply * reply) {
     (void)context;
-    (void)reply;
+
+
+    if (!request->has_msg()) {
+        loge("PublishMqtt: message is missing\n");
+        return Status(StatusCode::INVALID_ARGUMENT, "message is missing");
+    }
+
+    const Mqtt5Message & message = request->msg();
+    MqttQoS qos = message.qos();
+    if (qos < QOS_MIN || qos > QOS_MAX) {
+        loge("PublishMqtt: invalid QoS %d, must be in range [%d,%d]\n", (int)qos, QOS_MIN, QOS_MAX);
+        return Status(StatusCode::INVALID_ARGUMENT, "invalid QoS, must be in range [0,3]");
+    }
+
+    const std::string & topic = message.topic();
+    if (topic.empty()) {
+        loge("PublishMqtt: topic is empty\n");
+        return Status(StatusCode::INVALID_ARGUMENT, "topic is empty");
+    }
+
+    int timeout = request->timeout();
+    if (timeout < TIMEOUT_MIN) {
+        loge("PublishMqtt: invalid timeout, must be at least %u second\n", TIMEOUT_MIN);
```

Comment:
```suggestion
        loge("PublishMqtt: invalid timeout, must be at least %u minute(s)\n", TIMEOUT_MIN);
```

Reply:
TIMEOUT_MIN is actually 1 seconds not minutes.

Reply:
ah, I misinterpred MIN as minutes :)